// slotify.js/dist/simulation/ReelGameWithFreeGamesSessionSimulationModel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReelGameWithFreeGamesSessionSimulationModel = void 0;
var GameSessionSimulationModel_1 = require("./GameSessionSimulationModel");
var ReelGameWithFreeGamesSessionSimulationModel = /** @class */ (function () {
    function ReelGameWithFreeGamesSessionSimulationModel(session) {
        this._session = session;
        this._baseSession = new GameSessionSimulationModel_1.GameSessionSimulationModel(session);
    }
    ReelGameWithFreeGamesSessionSimulationModel.prototype.getRtp = function () {
        return this._baseSession.getRtp();
    };
    ReelGameWithFreeGamesSessionSimulationModel.prototype.getTotalBetAmount = function () {
        return this._baseSession.getTotalBetAmount();
    };
    ReelGameWithFreeGamesSessionSimulationModel.prototype.getTotalReturnAmount = function () {
        return this._baseSession.getTotalReturnAmount();
    };
    ReelGameWithFreeGamesSessionSimulationModel.prototype.updateTotalBetBeforePlay = function () {
        if (!this._session.getFreeGameSum()) {
            this._baseSession.updateTotalBetBeforePlay();
        }
    };
    ReelGameWithFreeGamesSessionSimulationModel.prototype.updateTotalReturnAfterPlay = function () {
        this._baseSession.updateTotalReturnAfterPlay();
    };
    return ReelGameWithFreeGamesSessionSimulationModel;
}());
exports.ReelGameWithFreeGamesSessionSimulationModel = ReelGameWithFreeGamesSessionSimulationModel;
